### PR TITLE
Handle fileopen if running

### PR DESCRIPTION
**Pull request by HofiOne**
## Description

Added the following features:
- accept media file open requests and replace the current playlist with a new one even if the app already started
- two new Finder service items, "Enqueue next in Kodi", "Enqueue last in Kodi" to modify the current playlists without replacing it
- ability to accept file drag and drop onto application icon
- listing Kodi in "Open with..." Finder context menu for common media file types

## Motivation and Context

Kodi Mac does not allow modifying or replacing the current playlist once the app has started.

## Types of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/cores/VideoPlayer/VideoRenderers/LinuxRendererGL.cpp [hunk 1]
```diff
@@ -303,7 +303,7 @@ bool CLinuxRendererGL::Configure(unsigned int width, unsigned int height, unsign
   // on osx 10.9 mavericks we get a strange ripple
   // effect when rendering with pbo
   // when used on intel gpu - we have to quirk it here
-  if (CDarwinUtils::IsMavericks())
+  if (CDarwinUtils::IsMavericksOrHigher())
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/VideoRenderers/LinuxRendererGL.cpp]: this not not in category "non breaking change". you disable pbo for current OS versions which has major impact on performance.

Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/VideoRenderers/LinuxRendererGL.cpp]: you just changed the name of the method and revealed an issue. nice

Comment (HofiOne) [inline xbmc/cores/VideoPlayer/VideoRenderers/LinuxRendererGL.cpp]: please explain how it breaks anything?

Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/VideoRenderers/LinuxRendererGL.cpp]: @HofiOne sorry for confusion, see my seconds post. By giving this method the correct name, you brought an already existing issue to surface. This is good! :)

Comment (HofiOne) [inline xbmc/cores/VideoPlayer/VideoRenderers/LinuxRendererGL.cpp]: yepp, that was the purpose I've renamed it :)
In a next round I will add the correct OSX version detection to CDarwinUtils, after that we should fix the current codes counting on a given exact OS version

**Comment by Rechi** [comment]
@HofiOne please remove the merge commits from this PR.
If your branch has conflicts with the master branch rebase onto it instead of resolving the conflicts during merging.

**Comment by Razzeee** [comment]
@HofiOne Any chance for a rebase?

Ping @Memphiz as this touches darwin/sdl

**Comment by Memphiz** [comment]
Please ping me again once this is merged. I need to backport that whole stuff to the removesdl branch once i have time for it again. Beside that i have no real opinion. 

Looks like this whole feature doesn't influence normal kodi behavior. In case people use that new feature and something doesn't work - i count on @HofiOne to be around to fix remaining issues.
I though wonder if there is no CFRelease needed somewhere in that new method "fileListFromPasteboard" ... 

We have no ARC - just fyi ...

**Comment by HofiOne** [comment]
@Memphiz, @Razzeee  my problem is that on this branch  happened changes by me and merging from master onto it multiple times in turn.
Does a simple rebase will solve this? It will have unresolved conflicts also, wont't it?

**Comment by Rechi** [comment]
Instead of `git merge upstream/master` do `git rebase upstream/master` and you have to resolve the conflicts. After you are finished with rebasing force push and the conflicts should be gone.

**Comment by HofiOne** [comment]
@Memphiz where do you miss a CFRelease exactly in fileListFromPasteboard ?
as I can see all the allocated objects are handled fine, but i maybe miss something

**Comment by Rechi** [comment]
@HofiOne seems you still haven't managed to rebase this correctly. If you need some help you can ping me at #kodi-dev on [freenode](//freenode.net) or open a thread in the kodi development subforum.

**Comment by HofiOne** [comment]
@Rechi , I've made a rebase, it led to one EMPTY conflict, i "resolved" that conflict, that's all what happened.

**Comment by Rechi** [comment]
Github still shows `This branch has conflicts that must be resolved`, so you haven't fully succeed in rebasing this onto latest master.

**Comment by HofiOne** [comment]
@Rechi i hope this time it will be ok

**Comment by Rechi** [comment]
@HofiOne nope you added a merge commit again instead rebasing it correctly on latest master.

**Comment by Memphiz** [comment]
This PR has now 250 commits and more ... wanted to have a look again on the objective-c reference counting - will wait until the rebate is sorted out

**Comment by HofiOne** [comment]
ok let's do the following.
let's drop this pull request currently, i will re-add my changes after @Memphiz finished the SDL remove
why:
- i'm not sure if the resolved conflicts are all fine as currently i cannot test it, i'm unable to build kodi on my 10.13 box neither with xcode 8.3.3 nor with 9.0 because some kind of asm compilation problem
- it would be easier for me to re-add these changes than @Memphiz to take care of these too during  the SDL removal
- it took too much effort and might will take more to make this fine, it would cost less to re-add the changes again

apart from that, i'd be curious what am i doing wrong with the rebase.
i switch to master, pulled all of the latest changes after that i'was making the rebase, resolved all the conflicts step by step at each commit attempt and continuing the rebase till all the conflicts passed away, I'm sure till this point everything was fine

but after i finished to re-apply all my changes on top of the master with the rebase, there had been a stage when i had to pull again to be able push mine, that is what i do not really understand, why it happened, there should have been a clean state where i only should have to commit and pussh my changes, but not that heppened

**Comment by Rechi** [comment]
Maybe your issue was you didn't force push (https://git-scm.com/docs/git-push#git-push--f).
Can you try again to rebase on latest master and force push to your `Handle-fileopen-if-running` branch.

**Comment by Memphiz** [comment]
@HofiOne please don't rely on my work here. The removesdl project is stuck again due to the fact that i have no time for working on it.

**Comment by HofiOne** [comment]
created a new branch for this, added all the changes in one run

https://github.com/xbmc/xbmc/pull/12821

closing this one

**Pull request closed without merging**